src/sci/impl/types.cljc

```diff
@@ -10,16 +10,17 @@
   (getVal [_this] v))
 
 (defprotocol IReified
-  (getInterface [_])
+  (getInterfaces [_])
   (getMethods [_]))
 
-(deftype Reified [interface meths]
+(deftype Reified [interfaces meths]
   IReified
-  (getInterface [_] interface)
+  (getInterfaces [_] interfaces)
   (getMethods [_] meths))
 
 (defn type-impl [x & _xs]
-  (or (when (instance? sci.impl.types.Reified x)
+  (or (when #?(:clj (instance? sci.impl.types.IReified x)
+               :cljs (clojure.core/satisfies? IReified x))
```

Comment:
@GreshamDanielStephens Why was this change needed?

Reply:
to be honest this confused me, and I'm pretty open to a better solution.
This stems from lack of clojureScript understanding I expect.
I had a test failure when using just `instance?`, I added a println for debugging and this is what I got.
```
(println x (instance? Reified x) (instance? IReified x) (satisfies? IReified x))

#object[sci.impl.types.Reified] true false true
```
So here I see in clojureScript I have an object of type Reified, which is an instance of Reified, but not of it's protocol IReified.
I started searching how to check it implements the protocol and found `satisfies?`
I assume this is a difference in how javascript and java work, and to do with IReified being an interface.

Testing again, satisfies looks like it works for both so I can go with that.

Reply:
In the spirit of only making changes that are necessary, why don't leave it like it was?

`(instance? sci.impl.types.Reified x)` for both cases?

`satisfies?` is a slow operation in Clojure(Script)

Reply:
`(instance? sci.impl.types.Reified x)` won't work for the new clj implementation of reify, since it directly implements IReified again without touching Reified. Didn't know satisfies is slow, I guess I can wrap just the class bit in the case `(instance? #?(:clj sci.impl.types.IReified :clj sci.impl.types.Reified) x)`?

Reply:
OK, let's do that (the second conditional should be `:cljs`)